Zend/tests/final_properties/inheritance/inheritance_success1.phpt

```diff
@@ -0,0 +1,19 @@
+--TEST--
+Test that final properties can be overwritten by non-final properties
+--FILE--
+<?php
+
+class Foo
+{
+    final protected static int $property1;
+    final protected int $property2;
+}
+
+class Bar extends Foo
+{
+    protected static int $property1;
+    protected int $property2;
```

Comment:
I don't think the inheritance behavior is mentioned in the RFC yet. Dropping final does seem sound though.

I was also wondering if we could make the type covariant for readonly properties. I.e. `final iterable $iter` to `final array $iter`. Technically this is not quite sound because it restricts the type of the initial assignment...

Reply:
Unfortunately, I neglected the inheritance aspect because I concentrated too much on making runtime restrictions work... So I'll add a related paragraph to the RFC.

Would property covariance be special to readonly properties? For BC reasons maybe? (properties are now invariant, right?)